src/Utilities/FlowAnalysis/FlowAnalysis/Analysis/GlobalFlowStateAnalysis/GlobalFlowStateDataFlowOperationVisitor.cs

```diff
@@ -61,142 +57,18 @@ private static AnalysisEntity GetGlobalEntity(GlobalFlowStateAnalysisContext ana
                 parent: null);
         }
 
-        public sealed override GlobalFlowStateAnalysisData Flow(IOperation statement, BasicBlock block, GlobalFlowStateAnalysisData input)
+        protected TAbstractAnalysisValue GlobalState
         {
-            EnsureInitialized(input);
-            return base.Flow(statement, block, input);
+            get => GetAbstractValue(GlobalEntity);
+            set => SetAbstractValue(GlobalEntity, value);
         }
 
-        private void EnsureInitialized(GlobalFlowStateAnalysisData input)
```

Comment:
Why can't this method be retained here with `TAnalysisData` as the parameter? Similarly, can we do so for other methods in this file and avoid duplication in both derived types?

Reply:
The reason is because `TAnalysisData : DictionaryAnalysisData<AnalysisEntity, TAbstractAnalysisValue>` can't be added.
So here we can't use the `ContainsKey` method

Reply:
As suggested above, we should just replace `TAnalysisData` with `DictionaryAnalysisData<AnalysisEntity, TAbstractAnalysisValue>` in this file and remove `TAnalysisData` type parameter.

Reply:
Resolved